src/masternodes/mn_checks.cpp

```diff
@@ -2674,9 +2675,71 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor
         if (!IsVaultPriceValid(mnview, obj.vaultId, height))
             return Res::Err("Cannot payback loan while any of the asset's price is invalid");
 
+        auto allowDFIPayback = false;
+        std::map<CAttributeType, CAttributeValue> attrs;
+        auto tokenDUSD = mnview.GetToken("DUSD");
+        if (tokenDUSD) {
+            const auto pAttributes = mnview.GetAttributes();
+            if (pAttributes) {
+                attrs = pAttributes->attributes;
+                CDataStructureV0 activeKey{AttributeTypes::Token, tokenDUSD->first.v, TokenKeys::PaybackDFI};
+                try {
+                    const auto& value = attrs.at(activeKey);
+                    auto valueV0 = boost::get<const CValueV0>(&value);
+                    if (valueV0) {
+                        const auto active = boost::get<const bool>(valueV0);
+                        if (active || *active) {
```

Comment:
I assume that this has some logic why it is written that way. but its super hard to read and looks fishy. 🤔

Reply:
Good spot, || is a bug. Updated.